WikiWatershed/model-my-watershed
# Adjust CSV downloads

## Overview

Replace the custom CSV download with the built-in export functionality of bootstrap-table. CSVs will now match their corresponding table in the UI exactly.

A slight hack is required to render tables with pagination. An additional table with the full dataset is rendered beneath the paginated table. It's then used to generated the CSV and then removed
from the DOM.

Connects #2032 

### Demo

 Attached is a zip containing an export of all the tables that can be downloaded as a CSV.
[csv_exports.zip](https://github.com/WikiWatershed/model-my-watershed/files/1149424/csv_exports.zip).

Here is the animal table, as an example.

![image](https://user-images.githubusercontent.com/1042475/28226645-ea4c0ff0-68a4-11e7-88bd-67212382f25a.png)

|                      |       | 
|----------------------|-------| 
| Animal               | Count | 
| "Chickens, Broilers" | 138   | 
| "Chickens, Layers"   | 0     | 
| "Cows, Beef"         | 0     | 
| "Cows, Dairy"        | 9     | 
| Horses               | 4     | 
| Pigs/Hogs/Swine      | 13    | 
| Sheep                | 1     | 
| Turkeys              | 26    | 

## Testing Instructions

- Create a new project with an AoI that is large enough to generate multiple pages for Water Quality and Point Source analysis tables. Something larger than Philadelphia should work.
- Select one of the models.
- Go to every table in the UI and download the data. Verify that the CSV matches the table in the UI.
- Repeat the process for the other model.
